Main.java

```diff
@@ -0,0 +1,50 @@
+import exceptions.DuplicateProductException;
+import exceptions.InvalidInputDataException;
+import exceptions.ProductNotFoundException;
+import exceptions.InsufficientQuantityException;
+import model.Product;
+import service.ProductManager;
+import java.util.Scanner;
+
+public class Main {
+    public static void main(String[] args) {
+        ProductManager productManager = new ProductManager();
+        Scanner scanner = new Scanner(System.in);
+
+        try {
+            // Taking input from user to add products
+            System.out.println("Enter product details:");
+            System.out.print("ID: ");
+            int id = scanner.nextInt();
+            scanner.nextLine(); // Consume newline character
+            System.out.print("Name: ");
+            String name = scanner.nextLine();
+            System.out.print("Price: ");
+            double price = scanner.nextDouble();
+            System.out.print("Quantity: ");
+            int quantity = scanner.nextInt();
+
+            productManager.addProduct(new Product(id, name, price, quantity));
+
+            // Updating product price
+            productManager.updateProduct(new Product(2, "Phone", 649.99, 20));
```

Comment:
why we are hardcoding these values and not taking them from user?

Reply:
Done